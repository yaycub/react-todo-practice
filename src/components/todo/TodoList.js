import React from 'react';
import PropTypes from 'prop-types';
import TodoItem from './TodoItem';
import styles from './styles/TodoItem.css';

const TodoList = ({ todos, setTodos }) => {
  const handleDelete = (name) => {
    setTodos(state => state.filter(todos => todos.name !== name));
  };

  const todoItemElements = todos.map((todo, i) => (
    <li key={i}>
      <TodoItem 
        name={todo.name}
        completed={todo.completed}
        handleDelete={handleDelete}
        styles={styles}
        index={i}
      />
    </li>
  ));

  return (
    <ul>
      Todos:
      {todoItemElements}
    </ul>
  );
};

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  setTodos: PropTypes.func.isRequired
};

export default TodoList;
