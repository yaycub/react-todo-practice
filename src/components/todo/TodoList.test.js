import React from 'react';
import { shallow } from 'enzyme';
import TodoList from './TodoList';

describe('TodoList Component', () => {
  it('can render TodoList', () => {
    const wrapper = shallow(<TodoList todos={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
