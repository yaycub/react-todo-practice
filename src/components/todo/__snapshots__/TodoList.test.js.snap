// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Component can render TodoList 1`] = `
<ul>
  Todos:
</ul>
`;
